File: 1158/1158.py
def insertion_sort(unsoroted_list):
    j = 1
    for j in range(j, len(unsoroted_list)):
        key = unsoroted_list[j]
        i = j - 1
        while i >= 0 and unsoroted_list[i] > key:
            unsoroted_list[i + 1] = unsoroted_list[i]
            i = i - 1
        unsoroted_list[i + 1] = key
        print(' '.join(unsoroted_list))

    return unsoroted_list
num = int(input())
unsoroted_list = input().split(' ')
sorted_list = insertion_sort(unsoroted_list)

File: jungol/1158/1158.c
#include <stdio.h>
#include <string.h>
void insertion_sort(int unsorted_list)
{
	int i,j;
	for(j = 0; j < strlen(unsorted_list); j++)
	{
		int key = unsorted_list[j];
		i = j - 1;
		while (i >= 0 && unsorted_list[i] > key)
		{
			unsorted_list[i + 1] = unsorted_list[i]
			i = i - 1;
		}
		unsorted_list[i + 1] = key
		printf("%d ",unsorted_list)
	}
	return unsorted_list;
}
int main()
{
	int num;
	scanf("%d",&num);
	int unsorted_list[100] = {0};
	for (i = 0; i < num; i++)
	{
		scanf("%d",unsorted_list[i]);
	}
	int sorted_list = insertion_sort(unsorted_list);
}
